### PR TITLE
Documentation for AddElement, Logical, and Resource

### PR DESCRIPTION
**Pull request by cmoesel**
Documentation for the new logical model and resource capabilities.

Because AddElement rule has so many possible variations, I struggled w/ the sections defining possible syntaxes.  I've very open to suggestions there (as long as they are consistent w/ the approach to syntax already practiced in the spec.

I also am not sure what formatting style we want to take in the examples.  Most notably, should we try to keep AddElement rules to a single line when possible? Or should we always put the definition on a newline? Or should we always put short and definition on a newline?  What we do may influence the style of others.  The current PR tries to keep it to one line except when that is clearly too long.

Keeping to one line at all times:
```
Logical:        EmergencyVehicle
Title:          "Emergency Vehicle"
Description:    "An emergency vehicle, such as an ambulance or fire truck."
* identifier 0..* SU Identifier "Identifier(s) of the vehicle" "Vehicle identifiers may include VINs and serial numbers."
* make 0..1 SU Coding "The vehicle make" "The vehicle make, e.g., Chevrolet."
* make from EmergencyVehicleMake (extensible)
* model 0..1 SU Coding "The vehicle model" "The vehicle model, e.g., G4500."
* model from EmergencyVehicleModel (extensible)
* year 0..1 SU positiveInteger "Year of manufacture" "The year the vehicle was manufactured"
* servicePeriod 0..1 Period "When the vehicle was in service" "Start date and end date (if applicable) when the vehicle operated."
* operator 0..* Reference(Organization or Practitioner or PractitionerRole) "The operator" "The organization or persons repsonsible for operating the vehicle"
* device 0..* Reference(Device) "Devices on board" "Devices on board the vehicle."
```

Always putting definition on its own line:
```
Logical:        EmergencyVehicle
Title:          "Emergency Vehicle"
Description:    "An emergency vehicle, such as an ambulance or fire truck."
* identifier 0..* SU Identifier "Identifier(s) of the vehicle"
    "Vehicle identifiers may include VINs and serial numbers."
* make 0..1 SU Coding "The vehicle make"
    "The vehicle make, e.g., Chevrolet."
* make from EmergencyVehicleMake (extensible)
* model 0..1 SU Coding "The vehicle model"
    "The vehicle model, e.g., G4500."
* model from EmergencyVehicleModel (extensible)
* year 0..1 SU positiveInteger "Year of manufacture"
    "The year the vehicle was manufactured"
* servicePeriod 0..1 Period "When the vehicle was in service"
    "Start date and end date (if applicable) when the vehicle operated."
* operator 0..* Reference(Organization or Practitioner or PractitionerRole) "The operator"
    "The organization or persons repsonsible for operating the vehicle"
* device 0..* Reference(Device) "Devices on board"
    "Devices on board the vehicle."
```

Always putting short and definition on their own lines:
```
Logical:        EmergencyVehicle
Title:          "Emergency Vehicle"
Description:    "An emergency vehicle, such as an ambulance or fire truck."
* identifier 0..* SU Identifier
    "Identifier(s) of the vehicle"
    "Vehicle identifiers may include VINs and serial numbers."
* make 0..1 SU Coding
    "The vehicle make"
    "The vehicle make, e.g., Chevrolet."
* make from EmergencyVehicleMake (extensible)
* model 0..1 SU Coding
    "The vehicle model"
    "The vehicle model, e.g., G4500."
* model from EmergencyVehicleModel (extensible)
* year 0..1 SU positiveInteger
    "Year of manufacture"
    "The year the vehicle was manufactured"
* servicePeriod 0..1 Period
    "When the vehicle was in service"
    "Start date and end date (if applicable) when the vehicle operated."
* operator 0..* Reference(Organization or Practitioner or PractitionerRole)
    "The operator"
    "The organization or persons repsonsible for operating the vehicle"
* device 0..* Reference(Device)
    "Devices on board"
    "Devices on board the vehicle."
```

Now that I've seen the comparison directly, I kind of like the last option (short and definition on their own lines), but I'll leave it as-is for now to get your feedback.

### CODE REVIEW

File: input/pagecontent/overview.md [hunk 1]
```diff
@@ -123,7 +125,18 @@ The keyword section is followed by a number of rules. Rules are the mechanism fo
 * {rule statement}
 ```
 
-The [formal syntax of rules](reference.html#rules-for-profiles-extensions-and-instances) are given in the [FSH Language reference](reference.html). Here is a summary of some of the more important rules in FSH:
+The [formal syntax of rules](reference.html#rules-for-profiles-extensions-logical-models-resources-and-instances) are given in the [FSH Language reference](reference.html). Here is a summary of some of the more important rules in FSH:
+
+* **AddElement rules** are used to define new elements in logical models and resources. They specify a path, cardinality, type(s), short definition, and optional long definition. For example:
```
Comment (joepaquette) [inline input/pagecontent/overview.md:130]: The optional flags is missing

File: input/pagecontent/reference.md [hunk 2]
```diff
@@ -2114,6 +2197,42 @@ Invariants are defined using the keywords `Invariant`, `Description`, `Expressio
...
+  Title:          "Human Being"
+  Description:    "A member of the Homo sapien species."
+  * name 0..* SU HumanName "Name(s) of the human" "The names by which the human is or has been known"
+  * birthDate 0..1 SU dateTime "The date of birth, if known"
+      "The date on which the person was born. Approximations may be used if exact date is unknown."
+  * deceased[x] 0..1 SU boolean or dateTime or Age "Indication if the human is deceased"
+      "An indication if the human has died. Boolean should not be used if date or age at death are known."
+  * family BackboneElement 0..1 "Family" "Members of the human's immediate family."
+    * mother 0..2 FamilyMember "Mother" "Biological mother, current adoptive mother, or both."
+    * mother 0..2 FamilyMember "Father" "Biological father, current adoptive father, or both."
```
Comment (joepaquette) [inline input/pagecontent/reference.md:2225]: path should be father

**Comment by joepaquette** [comment]
Regarding your question on formatting examples... I don't have a strong opinion one way or another since users are free to manage whitespace as desired. That said, if I had to pick one, I'd probably pick the middle option where the main definition is on one line and the definition indented underneath. This introduces some visual separation among AddElementRules while not making the example too long. But again, I'm fine with any of the options.

**Comment by cmoesel** [comment]
Fixed the two issues reported by @joepaquette.  Thanks, Joe!

Will wait on changing overall format of Logical/Resource examples until @markkramerus weighs in on the topic.

File: input/pagecontent/reference.md [hunk 3]
```diff
@@ -2195,6 +2314,36 @@ To define a profile, the keywords `Profile` and `Parent` are required, and `Id`,
...
+Only [DomainResource](http://hl7.org/fhir/R4/domainresource.html)  and [Resource](http://hl7.org/fhir/R4/resource.html) are allowed as parents of a resource. If no `Parent` is specified, DomainResource is used as the default parent.
+
+Rules defining the resource follow immediately after the keyword section. Resources are primarily comprised of AddElement rules. Since resources are represented using StructureDefinition, many of the rules used in profile definitions may also be used when defining resources. According to the FHIR specification's [interpretation of ElementDefinition in different contexts](http://hl7.org/fhir/R4/elementdefinition.html#interpretation), however, resources may not use slicing or fixed/patterned values.  As a result, Contains rules and Assignment rules are forbidden in resource definitions. Resources are also prohibited from constraining elements inherited from a parent definition.
+
+**Example:**
+
+* Define a resource representing an emergency vehicle:
+
+  ```
+  Logical:        EmergencyVehicle
```
Comment (ngfreiter) [inline input/pagecontent/reference.md:2332]: Should this be `Resource: `?

Comment (cmoesel) [reply input/pagecontent/reference.md:2332]: Mmmm.  Maybe... 

Look!  A squirrel!  Over there!

Comment (cmoesel) [reply input/pagecontent/reference.md]: If I (hypothetically) were to use the `Logical` keyword in the only example I have of a `Resource`, I would (hypothetically) fix it in some commit like 8996a60.

File: input/pagecontent/reference.md [hunk 4]
```diff
@@ -2114,6 +2197,42 @@ Invariants are defined using the keywords `Invariant`, `Description`, `Expressio
...
+  ```
+  Logical:        Human
+  Title:          "Human Being"
+  Description:    "A member of the Homo sapien species."
+  * name 0..* SU HumanName "Name(s) of the human" "The names by which the human is or has been known"
+  * birthDate 0..1 SU dateTime "The date of birth, if known"
+      "The date on which the person was born. Approximations may be used if exact date is unknown."
+  * deceased[x] 0..1 SU boolean or dateTime or Age "Indication if the human is deceased"
+      "An indication if the human has died. Boolean should not be used if date or age at death are known."
+  * family BackboneElement 0..1 "Family" "Members of the human's immediate family."
```
Comment (ngfreiter) [inline input/pagecontent/reference.md:2223]: Looks like `BackboneElement` and `0..1` are out of order.

Comment (cmoesel) [reply input/pagecontent/reference.md:2223]: I knew I probably should have tried compiling the examples before documenting them.  But I thought "Nah, they're so simple.  What could you possibly screw up?"

Comment (cmoesel) [reply input/pagecontent/reference.md]: @ngfreiter -- fixed (along w/ a couple of other things) in daeda35.

**Comment by cmoesel** [comment]
I'll fix the conflicts now.

File: input/pagecontent/reference.md (hunk 3 above)
Comment (jafeltra) [reply input/pagecontent/reference.md:2443]: @cmoesel and @markkramerus it looks like the `Logical: ` keyword made its way back. Is that intentional? It doesn't look like there are any other examples for `Resource: ` in the documentation.

Comment (cmoesel) [reply input/pagecontent/reference.md:2443]: It looks like it somehow snuck back in in a merge commit.  I'll submit a PR to fix it back.

Comment (cmoesel) [reply input/pagecontent/reference.md:2443]: I think Mark and I were both making edits at the same time and some wires got crossed.  I'll submit a new PR to fix it.

**Pull request merged**